### PR TITLE
Add use case on proactive cost controls

### PR DESCRIPTION
**Pull request by chipzoller**
Signed-off-by: chipzoller <chipzoller@gmail.com>

## Proposed Changes

Adds a new page under the Using Kubecost section called "Proactive Cost Controls" which describes a use case on how to prevent cost overruns before they occur. This links to a Kubecost blog post. Further links on the predictions API page have been added and a new section to the budgets page to remain consistent with page styling.

### CODE REVIEW

**Comment by bstuder99** [comment]
Thank you for the content updates. To be honest, I think it needs some expansion to exist as its own technical doc; at the moment, I think the use cases described in here should be applied directly into the Budgets/Prediction API docs. What I think we could also use is a step-by-step walkthrough of the UI on our [Navigating the Kubecost UI](https://docs.kubecost.com/using-kubecost/navigating-the-kubecost-ui) page, which will bring greater attention to Budgets (if we feel users do not fully recognize its utility and how to access it), along with all other main UI pages

**Comment by thomasvn** [comment]
I really like this update, and feel like we should continue to do more of this! Thanks @chipzoller! It's great to have our technical docs point to our blog posts to expand on interesting use cases and examples.

Up to @bstuder99 to decide whether we should begin making more of an official home for these examples in the technical docs themselves.

**Comment by chipzoller** [comment]
> I think it needs some expansion to exist as its own technical doc

What would you like to see here in order to be acceptable?

File: using-kubecost/navigating-the-kubecost-ui/budgets.md
Comment (kwombach12) [inline using-kubecost/navigating-the-kubecost-ui/budgets.md:66]: wording a tad awkward. Since this is under 'use cases' i'd suggest simply saying: "Enforce Kubecost Budgets to prevent overruns before they occur" -- you could maybe make the whole use case link to the guide

**Review by kwombach12 — CHANGES_REQUESTED** [review]

**Comment by kwombach12** [comment]
> > I think it needs some expansion to exist as its own technical doc
> 
> What would you like to see here in order to be acceptable?

I _think_ what @bstuder99 is alluding to here is that it feels almost a bit backward to link from technical docs to a blog -- in general I feel like its typically the other way around. Not opposed to how we have it structured in this PR as I know the walkthrough in the blog is on the longer side.

File: using-kubecost/proactive-cost-controls.md
Comment (kwombach12) [inline using-kubecost/proactive-cost-controls.md:5]:
```
This monitoring and subsequent reporting require that the deployment exist and be running for sometime and, therefore, cost some amount of money.
``` 
Feels unnecessary

Comment (kwombach12) [inline using-kubecost/proactive-cost-controls.md:5]: Is it just new deployments that we can block? Can't we also block against changes to existing deployments that would overrun a budget?

**Comment by bstuder99** [comment]
Not necessarily, we've linked to blogs in the past to reference example use cases. I'm saying what is described in this doc should be more of a step-by-step tutorial or something more in-depth to constitute a tech doc, otherwise this content can just be added to the Budget doc to clarify the feature's purpose more

File: using-kubecost/proactive-cost-controls.md
Comment (kwombach12) [inline using-kubecost/proactive-cost-controls.md:7]: instead of 'solve this challenge', I would suggest 'enforce budgets'

File: using-kubecost/proactive-cost-controls.md
Comment (kwombach12) [inline using-kubecost/proactive-cost-controls.md:10]: this is a bit confusing to me if I'm a new user... Also, I'd suggest we say 'Note:' instead of 'Note that'

**Review by kwombach12 — DISMISSED** [review]

File: using-kubecost/proactive-cost-controls.md
Comment (chipzoller) [reply using-kubecost/proactive-cost-controls.md:5]: Reworded.

File: using-kubecost/proactive-cost-controls.md
Comment (chipzoller) [reply using-kubecost/proactive-cost-controls.md:7]: Changed.

File: using-kubecost/proactive-cost-controls.md
Comment (chipzoller) [reply using-kubecost/proactive-cost-controls.md:10]: Simplified.

File: using-kubecost/navigating-the-kubecost-ui/budgets.md
Comment (chipzoller) [reply using-kubecost/navigating-the-kubecost-ui/budgets.md:66]: Like this?

**Review by chipzoller — COMMENTED** [review]
A few things here:

1. I didn't want to regurgitate the entire series of steps covered in the blog post in the same level of detail. It seems unnecessary to do so when the blog already does a decent job of this and in a step-by-step manner.
2. It isn't uncommon at all for technical documentation to reference a blog post (from the same project/product) as a form of auxillary documentation. Reference, usage, and use case docs are all different forms of docs and linking out to a different form admist another is common.
3. Budgets exist to service more than just this use case. Mixing use case docs (i.e., this one) with usage docs (the budgets docs) or reference docs (the API docs) generally doesn't work well since they have different goals.

Glad to make additional changes requested by the docs maintainers or, alternatively, if you feel this just isn't appropriate, feel free to close this PR entirely.

**Comment by bstuder99** [comment]
It sounds like this is content that could be directly added into the Budgets/Prediction API doc; once the user has understood how Budgets works, they may wonder if there is a way to directly control consumption of the resource, which the basic feature can't do. Then we can link them to our blog post which contains a Kyverno tutorial, or write a separate tech doc for Gatekeeper. Ultimately I don't know if the current doc is really explaining a whole lot that the blog it links to doesn't. The note about single-cluster Kubecost installations is an important technical detail, but other than that, this reads like a blog post.

**Review by chipzoller — COMMENTED** [review]
Added new headings with links to this doc.

**Comment by chipzoller** [comment]
Thanks for the edits. Will you handle the merging or is that up to me?

**Comment by srpomeroy** [comment]
Once approved, you can merge when ready.

**Pull request merged**
